### PR TITLE
Implement Bearer authentication token

### PR DESCRIPTION
**Pull request by therealplato**
This PR protects all routes except `/command` with an authentication token.

Gin will look at the `Authentication` header and expect to see something like `Bearer 12345`.
It will compare the token with `env.AuthBearerToken` which defaults to the value of the env var `BROADWAY_AUTH_TOKEN`.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -80,12 +80,16 @@ func (s *Server) Init() {
 func (s *Server) setupHandlers() {
 	s.engine = gin.Default()
 	gin.SetMode(gin.ReleaseMode) // Comment this to use debug mode for more verbose output
+	// Define routes:
+	s.engine.POST("/command", s.postCommand)
+	s.engine.GET("/command", s.getCommand)
+	// Protect subsequent routes with middleware:
+	s.engine.Use(authMiddleware)
```
Comment (macat) [inline server/server.go]: This is awesome. I didn't know you can use a middleware only for some handlers.

**Comment by macat** [comment]
+1

**Comment by poopoothegorilla** [comment]
+1

**Comment by macat** [comment]
MR STORY EATER

**Pull request merged**